### ISSUE TITLE
sql.h should not include from iODBC/sqltypes.h but from sqltypes.h

### ISSUE THREAD

**Issue by PerMildner** [openlink/iODBC#17]
On OS X I get this error:
`cc ... -I/Library/Frameworks/iODBC.framework/Headers odbc.c -c
In file included from odbc.c:123:
/Library/Frameworks/iODBC.framework/Headers/sql.h:89:10: fatal error: 
      'iODBC/sqltypes.h' file not found
#include <iODBC/sqltypes.h>`

The problem is that `/Library/Frameworks/iODBC.framework/Headers/sql.h` does:
`#include <iODBC/sqltypes.h>
`
But `sqltypes.h` is in the same directory as `sql.h` so it should just include it using
`#include <sqltypes.h>
`
(or perhaps `#include "sqltypes.h"`).

Note that there is no directory 'iODBC' anywhere under `/Library/Frameworks/iODBC.framework` so adding other include paths will not help.

This makes it impossible to use the include files installed by the 3.52.12 installer on OS X.

**Comment by sandan** [comment]
I am also getting this error on OS X. I had to remove the iODBC prefix in the `include` statement in sql.h and in sqltypes.h. Is there a way this is supposed to work?

@PerMildner I think the iODBC prefix has something to do with the fact that `/usr/local/iODBC/include/sql.h` exists when you install iODBC on MAC.

**Comment by TallTed** [comment]
@PerMildner @sandan -- On what version(s) of macOS (a/k/a OS X, a/k/a Mac OS X) are you seeing this occur?

@smalinin @pkleef -- This appears to be related to, if not an effective duplicate of, #27 and #28 ... Please check?

**Comment by smalinin** [comment]
@sandan 
It isn't enough info for recreate your issue. 
What are you doing ?
I could compile iODBC SDK on my OSX host without any errors like above.

**Comment by PerMildner** [comment]
@smalinin This bug is not about compiling the iODBC SDK. It is about _using_ the iODBC SDK that was installed by the 3.52.12 installer.

As I described above, compiling against the framework is done by specifying the include folder with `-I/Library/Frameworks/iODBC.framework/Headers` but the headers that have been installed there are broken since they refer to paths that are invalid in the _installed_ iODBC framework. As @TallTed mentioned, it seems related to #27 and #27.

(I do not remember which version of OS X this was, but I do not think it matters.)

**Comment by smalinin** [comment]
@PerMildner @sandan 
You used a wrong command for compile, I think. If you want to use *Frameworks*, you must to use OSX specific options or Xcode project. For example, we will build  [iodbctest.c](https://github.com/openlink/iODBC/blob/develop/samples/iodbctest.c)
- with Frameworks
```
% cc iodbctest.c -F /Library/Frameworks/iODBC.framework -framework iODBC  -o iodbctest
```
- without Frameworks (The Unix style iODBC libs, includes are installed to /usr/local/iODBC by default)
```
% cc iodbctest.c -I/usr/local/iODBC -DNO_FRAMEWORKS -L/usr/local/iODBC/lib -liodbc -o iodbctest
```

**Comment by PerMildner** [comment]
@smalinin Thanks! Apparently there is some magic happening with `#include <iODBC/sqltypes.h>` when a framework has been specified, so the `iODBC/`-part is discarded. See [the Apple documentation](https://developer.apple.com/library/content/documentation/MacOSX/Conceptual/BPFrameworks/Tasks/IncludingFrameworks.html). I had to add an explicit `-I/Library/Frameworks/iODBC.framework/Headers` to your example, but other than that your suggestion works.

So, the real underlying cause is a lack of documentation, i.e. there should be something that answers the question: "How to build iodbctest, on macOS, using the installed iODBC framework?"

For reference, the following transcript shows a build on a clean macOS 10.13.4, where I installed the "Command line tools", the 3.52.12 iODBC installer, and successfully built the iodbctest.c code:

```
# Prior to this I ran the cc command, which brings up a prompt where I chose to install the Command line tools.
#
$ cd
$ curl https://raw.githubusercontent.com/openlink/iODBC/develop/samples/iodbctest.c > iodbctest.c
$ cc iodbctest.c ## <<< As expected, this does not work (iODBC not installed yet)
iodbctest.c:83:10: fatal error: 'sql.h' file not found

$ curl http://opldownload.s3.amazonaws.com/uda/components/7.0/universal-apple-macosx10.7-32/mxkozzzz.dmg > mxkozzzz.dmg
$ open mxkozzzz.dmg 
$ open /Volumes/iODBC-SDK-3.52.12-MacOSX-10.7/iODBC-SDK.pkg ## << I then clicked through the installer, using the default choices.

$ cc iodbctest.c ## <<< As expected, this still does not work (iODBC framework/headers not specified to cc)
iodbctest.c:83:10: fatal error: 'sql.h' file not found

$ cc iodbctest.c -F /Library/Frameworks/iODBC.framework -framework iODBC  -o iodbctest ## <<< Still does not work (no -I... specified)
iodbctest.c:83:10: fatal error: 'sql.h' file not found

## Finally, success!
$ cc iodbctest.c -F /Library/Frameworks/iODBC.framework -framework iODBC -I/Library/Frameworks/iODBC.framework/Headers -o iodbctest
## (These warnings look like bugs in the iodbctest.c code)
iodbctest.c:945:20: warning: field width should have type 'int', but argument has type 'size_t' (aka 'unsigned long') [-Wformat]
              printf ("%-*.*s", displayWidth, displayWidth, colName);
                       ~~~^~~   ~~~~~~~~~~~~
iodbctest.c:1016:17: warning: field width should have type 'int', but argument has type 'size_t' (aka 'unsigned long') [-Wformat]
                  printf ("%-*.*s", displayWidths[colNum - 1],
                           ~~~^~~   ~~~~~~~~~~~~~~~~~~~~~~~~~
2 warnings generated.
$ ./iodbctest 
iODBC Demonstration program
This program shows an interactive SQL processor
Driver Manager: 03.52.1216.0526

Enter ODBC connect string (? shows list): ^C
$ uname -a
Darwin SPs-Mac.local 17.5.0 Darwin Kernel Version 17.5.0: Mon Mar  5 22:24:32 PST 2018; root:xnu-4570.51.1~1/RELEASE_X86_64 x86_64
$ cc --version
Apple LLVM version 9.1.0 (clang-902.0.39.1)
Target: x86_64-apple-darwin17.5.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
$ 
```

Thanks for the help!

**Comment by smalinin** [comment]
@PerMildner 
Ok, Yes the C behavior was changed in last in new versions of C compiler.
The iodbctest.c is Unix style app, so it is better to use:
```
% cc iodbctest.c -I/usr/local/iODBC -DNO_FRAMEWORKS -L/usr/local/iODBC/lib -liodbc -o iodbctest
```
and for last version of C compiler and with Frameworks, you must to use:
```
#include <iODBC/sql.c>
...
```
and compile command:
```
% cc iodbctest.c -F/Library/Frameworks/iODBC.framework -framework iODBC  -o iodbctest
```
It is more properly way.

**Comment by PerMildner** [comment]
Thanks. I can not use the `-I/usr/local/iODBC/` etc since that is not what the iODBC installer installs. I want to use the installer.

I do not think the C compiler has changed how it handles framework includes. The Apple-documentation I linked to was last updated 2013.

**Comment by TallTed** [comment]
@PerMildner - I'm pretty sure that you'll find the iODBC SDK installer does indeed deliver `/usr/local/iODBC/{bin,include,lib}` as well as the Frameworks.

**Comment by PerMildner** [comment]
TL;DR; The bug is real but it does not affect very recent macOS/clang-combinations.

@TallTed Your are right, of course. The 3.52.12 installer does install in `/usr/local/iODBC` as well.

However, the plot thickens. In `/usr/local/iODBC/include/` there is no `iODBC` directory. So how is `#include <iODBC/sqltypes.h>` supposed to be found when building without telling the compiler to use the framework (which apparently does some automagic stripping of framework names, see above)?

Interestingly the "Unix way" works anyway, for no documented reason that I can find, on the macOS 10.13 and clang-902 that I used above:
```
$ cc -c iodbctest.c -I/usr/local/iODBC/include
iodbctest.c:945:20: warning: field width should have type 'int', but argument has type 'size_t'
      (aka 'unsigned long') [-Wformat]
              printf ("%-*.*s", displayWidth, displayWidth, colName);
                       ~~~^~~   ~~~~~~~~~~~~
iodbctest.c:1016:17: warning: field width should have type 'int', but argument has type 'size_t'
      (aka 'unsigned long') [-Wformat]
                  printf ("%-*.*s", displayWidths[colNum - 1],
                           ~~~^~~   ~~~~~~~~~~~~~~~~~~~~~~~~~
2 warnings generated.
$ uname -a
Darwin SPs-Mac.local 17.5.0 Darwin Kernel Version 17.5.0: Mon Mar  5 22:24:32 PST 2018; root:xnu-4570.51.1~1/RELEASE_X86_64 x86_64
bash-3.2$ cc --version
Apple LLVM version 9.1.0 (clang-902.0.39.1)
Target: x86_64-apple-darwin17.5.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
$ 
```
(I had to adjust the `-I` path from the example of @smalinin and `-DNO_FRAMEWORKS` makes no difference, I do not know what it was supposed to do, it is not used by the installed headers.)

*However:* None of the above works on older systems, e.g. the system I used when I first reported this bug:
```
$ cc iodbctest.c -F /Library/Frameworks/iODBC.framework -framework iODBC -I/Library/Frameworks/iODBC.framework/Headers -o iodbctest
In file included from iodbctest.c:83:
/Library/Frameworks/iODBC.framework/Headers/sql.h:89:10: fatal error: 'iODBC/sqltypes.h' file not found
#include <iODBC/sqltypes.h>
        ^
1 error generated.
$ cc iodbctest.c -I/usr/local/iODBC/include -DNO_FRAMEWORKS -L/usr/local/iODBC/lib -liodbc -o iodbctest
In file included from iodbctest.c:83:
/usr/local/iODBC/include/sql.h:89:10: fatal error: 'iODBC/sqltypes.h' file not found
#include <iODBC/sqltypes.h>
        ^
1 error generated.
$ uname -a
Darwin octopussy-vm-mac-os-x-10-11.local 15.6.0 Darwin Kernel Version 15.6.0: Mon Oct  2 22:20:08 PDT 2017; root:xnu-3248.71.4~1/RELEASE_X86_64 x86_64
$ cc --version
Apple LLVM version 7.3.0 (clang-703.0.31)
Target: x86_64-apple-darwin15.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
$
```

Presumably all these problems would go away, on both older and newer systems, if all `#include <iODBC/XXX>` were changed to `#include "XXX"`

**Comment by smalinin** [comment]
@PerMildner 
The syntax `#include <iODBC/sqltypes.h>` is used for OSX only projects, that compiled/links with Frameworks. And it must(may) be used for OSX style programms.

**Comment by PerMildner** [comment]
@smalinin 
* Note that the problematic files that use `#include <iODBC/sqltypes.h>` are the iODBC headers, like `sql.h` that the iODBC installer installs (`#include <iODBC/sqltypes.h>` is used *both* in `/Library/Frameworks/iODBC/Headers/` *and* in `/usr/local/iODBC/include/`, so there is nothing I, as an end-user, can do about it. So, if `#include <iODBC/sqltypes.h>` is the wrong thing to do, then it is a bug in the files installed by the iODBC installer.
* "The iodbctest.c is Unix style program." Yes, but it can not be built on older macOS/clang, not even with the "Unix style" command above. Note that the installer is supposed to be compatible with 10.7.x to 10.11.x but the installed headers actually does not work in any of those versions. The headers do work, apparently by chance, in 10.14. And, of course, macOS is certified Unix.
* "You must choose,... and select a properly include style": The problem is not my code, it is the headers installed by the iODBC installer, so I can not make this choice.
* "iODBCdemo": I do not use Objective C, and I do not build with Xcode. I use only autoconf/make/cc from the command line.

The changes to the iODBC include files that I suggested, would, as far as I can tell, fix the bug in a platform independent way.

**Comment by smalinin** [comment]
@PerMildner 
The same is on OSX 10.10 with older c compiler
```
$ uname -a
Darwin mac15-10.local 14.5.0 Darwin Kernel Version 14.5.0: Sun Sep 25 22:07:15 PDT 2016; root:xnu-2782.50.9~1/RELEASE_X86_64 x86_64
$ cc --version
Apple LLVM version 7.0.0 (clang-700.0.72)
Target: x86_64-apple-darwin14.5.0
Thread model: posix
$ cc iodbctest.c -I/usr/local/iODBC/include -DNO_FRAMEWORKS -L/usr/local/iODBC/lib -liodbc -o iodbctest
iodbctest.c:945:20: warning: field width should have type 'int', but argument has type 'size_t' (aka 'unsigned long') [-Wformat]
              printf ("%-*.*s", displayWidth, displayWidth, colName);
                       ~~~^~~   ~~~~~~~~~~~~
iodbctest.c:1016:17: warning: field width should have type 'int', but argument has type 'size_t' (aka 'unsigned long') [-Wformat]
                  printf ("%-*.*s", displayWidths[colNum - 1],
                           ~~~^~~   ~~~~~~~~~~~~~~~~~~~~~~~~~
2 warnings generated.
```

**Comment by smalinin** [comment]
@PerMildner 
I wrote about very old compier **Apple LLVM version 4.2 (clang-425.0.28) , I think, that you don't have it, it used on OSX 10.7.

**Comment by smalinin** [comment]
@PerMildner 
The files in /usr/local/iODBC/include are links to files in Framework
```
$ cd /usr/local/iODBC/include
$ ls -la
total 96
drwxr-xr-x  14 root  wheel  476 May 23  2017 .
drwxr-xr-x   5 root  wheel  170 May 23  2017 ..
lrwxr-xr-x   1 root  wheel   52 May 23  2017 iodbc.h -> //Library/Frameworks/iODBC.framework/Headers/iodbc.h
lrwxr-xr-x   1 root  wheel   55 May 23  2017 iodbcext.h -> //Library/Frameworks/iODBC.framework/Headers/iodbcext.h
lrwxr-xr-x   1 root  wheel   60 May 23  2017 iodbcinst.h -> //Library/Frameworks/iODBCinst.framework/Headers/iodbcinst.h
lrwxr-xr-x   1 root  wheel   56 May 23  2017 iodbcunix.h -> //Library/Frameworks/iODBC.framework/Headers/iodbcunix.h
lrwxr-xr-x   1 root  wheel   51 May 23  2017 isql.h -> //Library/Frameworks/iODBC.framework/Headers/isql.h
lrwxr-xr-x   1 root  wheel   54 May 23  2017 isqlext.h -> //Library/Frameworks/iODBC.framework/Headers/isqlext.h
lrwxr-xr-x   1 root  wheel   56 May 23  2017 isqltypes.h -> //Library/Frameworks/iODBC.framework/Headers/isqltypes.h
lrwxr-xr-x   1 root  wheel   59 May 23  2017 odbcinst.h -> //Library/Frameworks/iODBCinst.framework/Headers/odbcinst.h
lrwxr-xr-x   1 root  wheel   50 May 23  2017 sql.h -> //Library/Frameworks/iODBC.framework/Headers/sql.h
lrwxr-xr-x   1 root  wheel   53 May 23  2017 sqlext.h -> //Library/Frameworks/iODBC.framework/Headers/sqlext.h
lrwxr-xr-x   1 root  wheel   55 May 23  2017 sqltypes.h -> //Library/Frameworks/iODBC.framework/Headers/sqltypes.h
lrwxr-xr-x   1 root  wheel   55 May 23  2017 sqlucode.h -> //Library/Frameworks/iODBC.framework/Headers/sqlucode.h
``` 
so , may be, something is wrong with your installed files and you will need reinstall iODBC SDK

**Comment by PerMildner** [comment]
On my OS X 10.10.5, which did not previously have iODBC installed, i get the same problem. It has an older clang than your 10.10, but it does not seem likely that the clang-versions you happen to have does one thing, and the ones I use do something else since we have tried many versions.

Since I just installed iODBC it seems unlikely that there is someting wrong with the installed files. Especially since they obviously tries to include things from a directory (`iODBC/`) that does not exist anywhere.

```
bash-3.2$ curl http://opldownload.s3.amazonaws.com/uda/components/7.0/universal-apple-macosx10.7-32/mxkozzzz.dmg > mxkozzzz.dmg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 1206k  100 1206k    0     0   282k      0  0:00:04  0:00:04 --:--:--  282k
bash-3.2$ open mxkozzzz.dmg 
bash-3.2$ open /Volumes/iODBC-SDK-3.52.12-MacOSX-10.7/iODBC-SDK.pkg 
bash-3.2$ curl https://raw.githubusercontent.com/openlink/iODBC/develop/samples/iodbctest.c > iodbctest.c
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 24902  100 24902    0     0  53945      0 --:--:-- --:--:-- --:--:-- 53900
bash-3.2$ cc iodbctest.c -I/usr/local/iODBC/include -DNO_FRAMEWORKS -L/usr/local/iODBC/lib -liodbc -o iodbctest
In file included from iodbctest.c:83:
/usr/local/iODBC/include/sql.h:89:10: fatal error: 'iODBC/sqltypes.h' file not
      found
#include <iODBC/sqltypes.h>
         ^
1 error generated.
bash-3.2$ cc iodbctest.c -F /Library/Frameworks/iODBC.framework -framework iODBC -I/Library/Frameworks/iODBC.framework/Headers -o iodbctest
In file included from iodbctest.c:83:
/Library/Frameworks/iODBC.framework/Headers/sql.h:89:10: fatal error: 
      'iODBC/sqltypes.h' file not found
#include <iODBC/sqltypes.h>
         ^
1 error generated.
bash-3.2$ cc --version
Apple LLVM version 6.0 (clang-600.0.54) (based on LLVM 3.5svn)
Target: x86_64-apple-darwin14.5.0
Thread model: posix
bash-3.2$ uname -a
Darwin octopussy-vm-mac-os-x-10-10.local 14.5.0 Darwin Kernel Version 14.5.0: Thu Apr 21 20:40:54 PDT 2016; root:xnu-2782.50.3~1/RELEASE_X86_64 x86_64
bash-3.2$ grep sqltypes.h /usr/local/iODBC/include/sql.h
#include <iODBC/sqltypes.h>
bash-3.2$ 
```
What does your 10.10 machine say if you try:
```
$ grep sqltypes.h /usr/local/iODBC/include/sql.h
$ grep sqltypes.h /Library/Frameworks/iODBC.framework/Headers/sql.h
$ ls -la /Library/Frameworks/iODBC.framework/Headers/
```

**Comment by smalinin** [comment]
@PerMildner 
https://developer.apple.com/library/content/documentation/MacOSX/Conceptual/BPFrameworks/Tasks/InstallingFrameworks.html
...
For nearly all cases, installing your frameworks in /Library/Frameworks is the best choice. **Frameworks in this location are discovered automatically by the compiler at compile time and the dynamic linker at runtime.** 
...

**Comment by smalinin** [comment]
@PerMildner 
I have rechecked on clear vm host with OSX10.11 (I don't have clear OSX10.10 now).
I installed iODBC SDK from http://opldownload.s3.amazonaws.com/uda/components/7.0/universal-apple-macosx10.7-32/mxkozzzz.dmg and tried compile iodbctest.c 
I works for me without error. I will try recheck on clear OSX 10.10 later.

**Comment by PerMildner** [comment]
@smalinin Yes, I want to use the framework version, for the reasons you cite.

This gets even stranger. I installed Xcode 7 on my OS X 10.10, and that did not help:
```
# The beginning is the same as above, i.e. OS X 10.10 with clang-600/Xcode 6.1:
bash-3.2$ pwd
/Users/spdev
bash-3.2$ curl http://opldownload.s3.amazonaws.com/uda/components/7.0/universal-apple-macosx10.7-32/mxkozzzz.dmg > mxkozzzz.dmg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 1206k  100 1206k    0     0   282k      0  0:00:04  0:00:04 --:--:--  282k
bash-3.2$ open mxkozzzz.dmg 
bash-3.2$ open /Volumes/iODBC-SDK-3.52.12-MacOSX-10.7/iODBC-SDK.pkg 
bash-3.2$ curl https://raw.githubusercontent.com/openlink/iODBC/develop/samples/iodbctest.c > iodbctest.c
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 24902  100 24902    0     0  53945      0 --:--:-- --:--:-- --:--:-- 53900
bash-3.2$ cc iodbctest.c -I/usr/local/iODBC/include -DNO_FRAMEWORKS -L/usr/local/iODBC/lib -liodbc -o iodbctest
In file included from iodbctest.c:83:
/usr/local/iODBC/include/sql.h:89:10: fatal error: 'iODBC/sqltypes.h' file not
      found
#include <iODBC/sqltypes.h>
         ^
1 error generated.
bash-3.2$ cc iodbctest.c -F /Library/Frameworks/iODBC.framework -framework iODBC -I/Library/Frameworks/iODBC.framework/Headers -o iodbctest
In file included from iodbctest.c:83:
/Library/Frameworks/iODBC.framework/Headers/sql.h:89:10: fatal error: 
      'iODBC/sqltypes.h' file not found
#include <iODBC/sqltypes.h>
         ^
1 error generated.
bash-3.2$ cc --version
Apple LLVM version 6.0 (clang-600.0.54) (based on LLVM 3.5svn)
Target: x86_64-apple-darwin14.5.0
Thread model: posix
bash-3.2$ uname -a
Darwin octopussy-vm-mac-os-x-10-10.local 14.5.0 Darwin Kernel Version 14.5.0: Thu Apr 21 20:40:54 PDT 2016; root:xnu-2782.50.3~1/RELEASE_X86_64 x86_64
bash-3.2$ grep sqltypes.h /usr/local/iODBC/include/sql.h
#include <iODBC/sqltypes.h>
bash-3.2$ cc --version
Apple LLVM version 6.0 (clang-600.0.54) (based on LLVM 3.5svn)
Target: x86_64-apple-darwin14.5.0
Thread model: posix

#### Here I installed Xcode 7 (by dragging it into "/Applications", and overwriting the pre-existing 6.1 "Xcode.app")
bash-3.2$ cc --version


Agreeing to the Xcode/iOS license requires admin privileges, please re-run as root via sudo.

####  Here I started Xcode and accepted the license agreement. Now cc (in the same shell process) is clang-700, and the problem persists:
bash-3.2$ cc --version
Apple LLVM version 7.0.0 (clang-700.0.72)
Target: x86_64-apple-darwin14.5.0
Thread model: posix
bash-3.2$ cc iodbctest.c -I/usr/local/iODBC/include -DNO_FRAMEWORKS -L/usr/local/iODBC/lib -liodbc -o iodbctest
In file included from iodbctest.c:83:
/usr/local/iODBC/include/sql.h:89:10: fatal error: 'iODBC/sqltypes.h' file not
      found
#include <iODBC/sqltypes.h>
         ^
1 error generated.
bash-3.2$ cc iodbctest.c -F /Library/Frameworks/iODBC.framework -framework iODBC -I/Library/Frameworks/iODBC.framework/Headers -o iodbctest
In file included from iodbctest.c:83:
/Library/Frameworks/iODBC.framework/Headers/sql.h:89:10: fatal error: 
      'iODBC/sqltypes.h' file not found
#include <iODBC/sqltypes.h>
         ^
1 error generated.
bash-3.2$ find /System/Library -name iODBC 2>/dev/null
bash-3.2$ find /Library -name iODBC 2>/dev/null
/Library/Application Support/iODBC
/Library/Frameworks/iODBC.framework/iODBC
/Library/Frameworks/iODBC.framework/Versions/3.52/iODBC
bash-3.2$ 
bash-3.2$ env
SHELL=/bin/bash
TERM=dumb
TMPDIR=/var/folders/k2/zhysf48x5wl3_8d6b194db_40000gn/T/
Apple_PubSub_Socket_Render=/private/tmp/com.apple.launchd.7L4xPYvmm9/Render
USER=...
EMACS=t
SSH_AUTH_SOCK=/private/tmp/com.apple.launchd.vYMJCBelNb/Listeners
TERMCAP=
__CF_USER_TEXT_ENCODING=0x1F5:0x0:0x0
COLUMNS=80
PATH=/usr/bin:/bin:/usr/sbin:/sbin
PWD=/Users/...
XPC_FLAGS=0x0
XPC_SERVICE_NAME=0
SHLVL=1
HOME=/Users/...
LOGNAME=...
DISPLAY=/private/tmp/com.apple.launchd.ve9CcvUwiK/org.macosforge.xquartz:0
INSIDE_EMACS=24.4.1,comint
_=/usr/bin/env
```

So now I see a different behavior than you, with the same Xcode/clang (`clang-700.0.72`), and the same iODBC installer. The real mystery is why it works for you, since your non-framework build apperently managed to open the non-existing `iODBC/sqltypes.h`.

Could it be that you have some other `iODBC` that clang picks up (see the `find...` in my transcript).
Could there be something in your environment that affects clang?

**Comment by smalinin** [comment]
@PerMildner 
I could recreate your problem on clear installation with OSX 10.10.5 :
```
Darwin sergeis-Mac-2.local 14.5.0 Darwin Kernel Version 14.5.0: Wed Jul 29 02:26:53 PDT 2015; root:xnu-2782.40.9~1/RELEASE_X86_64 x86_64
```
I will recheck on another version OSX and etc and create something for resolve problem.

**Comment by lyndonb-bq** [comment]
@smalinin Any information on this?

**Comment by smalinin** [comment]
@lyndonb-bq 
The code may be compiled with command like on OSX 10.14 
```
% cc iodbctest.c -F/Library/Frameworks -I/usr/local/iODBC/include -framework iODBC  -o iodbctest
```

**Comment by smalinin** [comment]
Fix for files in /usr/local/iODBC/include will be commited soon.
